src/server/common.cc

```diff
@@ -318,8 +293,7 @@ OpResult<ScanOpts> ScanOpts::TryFrom(CmdArgList args) {
       if (scan_opts.pattern == "*")
         scan_opts.pattern = string_view{};
     } else if (opt == "TYPE") {
-      ToLower(&args[i + 1]);
-      scan_opts.type_filter = ArgS(args, i + 1);
+      scan_opts.type_filter = CompactObjTypeConverter::ObjTypeFromString(ArgS(args, i + 1));
```

Comment:
Maybe Redis warns actually 🤔

Reply:
Added Log to the `ObjTypeFromString` function